### PR TITLE
Mes 3651 adding cat be to WRTC page

### PR DESCRIPTION
**Pull request by amobarak**
## Description

Added cat BE preparation for WRTC page.

## Checklist

- [x] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [ ] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [x] Squashed commit contains the JIRA ticket number

### CODE REVIEW

**Review by ammarhaiderbjss — CHANGES_REQUESTED** [review]
Can we please change the base branch to be `mes-607-cat-be`. Also the PR title should mention that this is just the WRTC screen.

File: src/pages/waiting-room-to-car/cat-be/waiting-room-to-car.cat-be.page.html [hunk 1]
```diff
@@ -0,0 +1,52 @@
+<ion-header>
+  <practice-mode-banner *ngIf="isPracticeMode"></practice-mode-banner>
```
Comment (TomBillingtonUK) [inline src/pages/waiting-room-to-car/cat-be/waiting-room-to-car.cat-be.page.html]: remove

File: src/pages/waiting-room-to-car/cat-be/waiting-room-to-car.cat-be.page.ts [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+  tellMeQuestionOutcome$: Observable<string>;
+  tellMeQuestion$: Observable<TellMeQuestion>;
+}
+
+@IonicPage()
+@Component({
+  selector: 'waiting-room-to-car-cat-be-page',
+  templateUrl: 'waiting-room-to-car.cat-be.page.html',
+})
+export class WaitingRoomToCarCatBEPage extends BasePageComponent {
```
Comment (TomBillingtonUK) [inline src/pages/waiting-room-to-car/cat-be/waiting-room-to-car.cat-be.page.ts]: I think we are doing Be not BE - Can you double check

File: src/pages/waiting-room-to-car/cat-be/waiting-room-to-car.cat-be.page.ts [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+    this.store$.dispatch(new TellMeQuestionDrivingFault());
+  }
+
+  eyesightTestResultChanged(passed: boolean): void {
+    const action = passed ? new EyesightTestPassed() : new EyesightTestFailed();
+    this.store$.dispatch(action);
+  }
+
+  getDebriefPage() {
+    return CAT_B.DEBRIEF_PAGE;
```
Comment (TomBillingtonUK) [inline src/pages/waiting-room-to-car/cat-be/waiting-room-to-car.cat-be.page.ts]: need to change

File: src/pages/waiting-room-to-car/cat-be/waiting-room-to-car.cat-be.page.ts [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+  instructorRegistrationChanged(instructorRegistration: number) {
+    this.store$.dispatch(new InstructorRegistrationNumberChanged(instructorRegistration));
+  }
+
+  onSubmit() {
+    Object.keys(this.form.controls).forEach(controlName => this.form.controls[controlName].markAsDirty());
+    if (this.form.valid) {
+      this.navController.push(CAT_BE.TEST_REPORT_PAGE).then(() => {
+        // remove Waiting Room To Car Page
+        const view = this.navController.getViews().find(view => view.id === CAT_B.WAITING_ROOM_TO_CAR_PAGE);
```
Comment (TomBillingtonUK) [inline src/pages/waiting-room-to-car/cat-be/waiting-room-to-car.cat-be.page.ts]: Need to change to CAT_BE

File: src/pages/waiting-room-to-car/cat-be/waiting-room-to-car.cat-be.page.ts (hunk 2 above)
Comment (TomBillingtonUK) [inline src/pages/waiting-room-to-car/cat-be/waiting-room-to-car.cat-be.page.ts]: You haven't updated the CAT_BE Page Constants object with this new class

**Review by TomBillingtonUK — APPROVED** [review]
A few changes/missed things but overall looks good

File: src/pages/waiting-room-to-car/cat-be/waiting-room-to-car.cat-be.page.ts (hunk 2 above)
Comment (amobarak) [reply src/pages/waiting-room-to-car/cat-be/waiting-room-to-car.cat-be.page.ts]: Yup, we are :D, will change

**Review by MarkPatt89 — APPROVED** [review]

**Pull request merged**